tribits/core/package_arch/TribitsProjectWriteConfigFile.cmake

```diff
@@ -0,0 +1,228 @@
+  # we're building shared libraries.
+  if(BUILD_SHARED_LIBS)
+    string(REPLACE ";" ":" SHARED_LIB_RPATH_COMMAND
+     "${${PROJECT_NAME}_CONFIG_LIBRARY_DIRS}")
+    set(SHARED_LIB_RPATH_COMMAND ${CMAKE_SHARED_LIBRARY_RUNTIME_CXX_FLAG}${SHARED_LIB_RPATH_COMMAND})
+  endif()
+
+  # Custom code in configuration file.
+  set(PROJECT_CONFIG_CODE "")
+
+  # Appending the logic to include each package's config file.
+  set(LOAD_CODE "# Load configurations from enabled packages")
+  foreach(TRIBITS_PACKAGE ${FULL_PACKAGE_SET})
+    set(LOAD_CODE "${LOAD_CODE}
+include(\"${${TRIBITS_PACKAGE}_BINARY_DIR}/${TRIBITS_PACKAGE}Config.cmake\")")
+  endforeach()
+  set(PROJECT_CONFIG_CODE "${PROJECT_CONFIG_CODE}\n${LOAD_CODE}")
+
+  tribits_set_compiler_vars_for_config_file(INSTALL_DIR)
+
+  if (${PROJECT_NAME}_ENABLE_INSTALL_CMAKE_CONFIG_FILES)
+    # In TribitsProjectConfigTemplate.cmake.in, we would like to preserve
+    # ${}-variables after the conversion to TribitsProjectConfigTemplate.cmake.
+    # To this end, one typically uses the @-syntax for variables. That doesn't
+    # support nested variables, however.  Use ${PDOLLAR} as a workaround, cf.
+    # <http://www.cmake.org/pipermail/cmake/2013-April/054341.html>.
+    set(PDOLLAR "$")
+    set(tribitsInstallationDir
+      "${${PROJECT_NAME}_TRIBITS_DIR}/${TRIBITS_CMAKE_INSTALLATION_FILES_DIR}")
+    set(TRIBITS_PROJECT_INSTALL_INCLUDE_DIR "")
+    configure_file(
+      "${tribitsInstallationDir}/TribitsProjectConfigTemplate.cmake.in"
+      "${PROJECT_BINARY_DIR}/${PROJECT_NAME}Config.cmake" )
+  endif()
+
+  ######
+  # Create a configure file for the install tree and set the install target for it. This
+  # file isn't generally useful inside the build tree. It will be placed in the base
+  # install directory for ${PROJECT_NAME} when installed.
+  ######
+
+  # Set the include and library directories relative to the location
+  # at which the ${PROJECT_NAME}Config.cmake file is going to be
+  # installed. Note the variable reference below is escaped so it
+  # won't be replaced until a client project attempts to locate
+  # directories using the installed config file. This is to deal with
+  # installers that allow relocation of the install tree at *install*
+  # time.
+  # The export files are typically installed in
+  #     <install dir>/<lib path>/cmake/<package name>/.
+  # The relative path to the installation dir is hence k*(../) + ../../, where
+  # k is the number of components in <lib path>. Extract those here.
+  # This doesn't work if ${${PROJECT_NAME}_INSTALL_LIB_DIR} contains "./" or
+  # "../" components, but really, it never did. All of this should actually be
+  # handled by CMake's configure_package_config_file().
+  string(REPLACE "/" ";" PATH_LIST ${${PROJECT_NAME}_INSTALL_LIB_DIR})
+  set(RELATIVE_PATH "../..")
+  foreach(PATH ${PATH_LIST})
+    set(RELATIVE_PATH "${RELATIVE_PATH}/..")
+  endforeach()
```

Comment:
You can use `cmake_path(RELATIVE_PATH)` for this.

Reply:
I will use `cmake_path()`.  (We could not do that until we upgraded the min version of CMake in this PR since `cmake_path()` was only added in CMake 3.20.)

Reply:
Turns out that code was not even being used anymore.  I deleted it in commit 014a1538 (along with a very long comment written by the person that wrote this code) :-)